tedana/utils.py

```diff
@@ -90,19 +93,20 @@ def make_adaptive_mask(data, mask=None, getsum=False):
     masksum = (np.abs(echo_means) > lthrs).sum(axis=-1)
 
     if mask is None:
-        # make it a boolean mask to (where we have at least 1 echo with good signal)
-        mask = (masksum >= 3).astype(bool)
+        # make it a boolean mask to (where we have at least `threshold` echoes with good signal)
+        mask = (masksum >= threshold).astype(bool)
+        masksum[masksum < threshold] = 0
     else:
         # if the user has supplied a binary mask
         mask = load_image(mask).astype(bool)
         masksum = masksum * mask
-        mask = (masksum >= 3).astype(bool)
         # reduce mask based on masksum
         # TODO: Use visual report to make checking the reduced mask easier
-        if np.any(masksum[mask] < 3):
-            n_bad_voxels = np.sum(masksum[mask] < 3)
+        if np.any(masksum[mask] < threshold):
+            n_bad_voxels = np.sum(masksum[mask] < threshold)
             LGR.warning('{0} voxels in user-defined mask do not have good '
                         'signal. Removing voxels from mask.'.format(n_bad_voxels))
+            masksum[masksum < threshold] = 0
```

Comment:
I am not completely familiar with where the adaptive mask is used, but could adding the following line
```python
masksum[maskum == 1] = 2
```
simplify the downstream steps or would it break other parts of the code?

Reply:
I think it depends on where we want to bury the "treat one good echo as two good echoes" trick. The way it is now, we can dig into where the adaptive mask indicates only one good echo, and it's on the other functions to document that those voxels will be treated as if they have two good echoes (depending on the threshold). On the other hand, if we add that, then the other functions can be simplified a bit, but we won't be able to investigate the adaptive mask as well.

Reply:
That makes sense. Thanks.